akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/SlotState.scala

```diff
@@ -143,23 +149,23 @@ private[pool] object SlotState {
       ctx.debug("Connection attempt failed.")
       // FIXME: register failed connection attempt, schedule request for rerun, backoff new connection attempts
       ctx.closeConnection()
-      ctx.dispatchFailure(ongoingRequest, cause)
-      Unconnected
+      // FIXME: register failed connection attempt, backoff new connection attempts
+      WaitingForResponseDispatch(ongoingRequest, Failure(cause))
     }
 
     override def onConnectionFailed(ctx: SlotContext, cause: Throwable): SlotState = {
       ctx.debug("Connection failed.")
       // FIXME: register failed connection attempt, schedule request for rerun, backoff new connection attempts
       ctx.closeConnection()
-      ctx.dispatchFailure(ongoingRequest, cause)
-      Unconnected
+      // FIXME: register failed connection attempt, backoff new connection attempts
```

Comment:
ticketify?

Reply:
It's basically #1391 but I'll add it to the comment.